pkg/ca/googleca/v1/googleca.go

```diff
@@ -136,32 +136,40 @@ func Req(parent string, pemBytes []byte, cert *x509.Certificate) (*privatecapb.C
 	}, nil
 }
 
-func (c *CertAuthorityService) Root(ctx context.Context) ([]*x509.Certificate, error) {
-	c.cachedRootsOnce.Do(func() {
-		var pems string
-		var err error
-		cas := c.client.ListCertificateAuthorities(ctx, &privatecapb.ListCertificateAuthoritiesRequest{
-			Parent: c.parent,
-		})
-		for {
-			c, done := cas.Next()
-			if done == iterator.Done {
-				break
-			}
-			if done != nil {
-				break
-			}
-			pems += strings.Join(c.PemCaCertificates, "")
-		}
-		c.cachedRoots, err = cryptoutils.LoadCertificatesFromPEM(strings.NewReader(pems))
-		if err != nil {
-			panic(fmt.Errorf("failed parsing PemCACertificates: %w", err))
-		}
+func (c *CertAuthorityService) TrustBundle(ctx context.Context) ([][]*x509.Certificate, error) {
+	// if we've already successfully fetched the CA info, just use the cached value
+	if c.cachedRoots != nil {
+		return [][]*x509.Certificate{c.cachedRoots}, nil
+	}
+
+	// fetch the latest values for the specified CA
+	var pems string
+	cas := c.client.ListCertificateAuthorities(ctx, &privatecapb.ListCertificateAuthoritiesRequest{
+		Parent: c.parent,
 	})
-	if len(c.cachedRoots) == 0 {
-		return c.cachedRoots, fmt.Errorf("error fetching root certificates")
+	for {
+		ca, done := cas.Next()
+		if done == iterator.Done {
+			break
+		} else if done != nil {
+			// if the iterator returns an issue for some reason, exit
+			return [][]*x509.Certificate{}, done
+		}
+		pems += strings.Join(ca.PemCaCertificates, "")
```

Comment:
A CA pool can have multiple CAs. This is concatenating all of the CA certs in a CA pool together, so we instead should have list of chains (each chain represented as a string), then for each string, `cryptoutils.LoadCertificatesFromPEM` to create []*x509.Certificate, then append that to a [][]*x509.Certificate.